File: web/src/types/podcast.ts
export interface Podcast {
  id: string;
  library_id: string;
  feed_url: string;
  title: string;
  description?: string | null;
  cover_url?: string | null;
  author?: string | null;
  auto_download: boolean;
  last_checked_at?: string | null;
  episode_count?: number;
  created_at: string;
  updated_at: string;
}

export interface PodcastEpisode {
  id: string;
  podcast_id: string;
  guid: string;
  title: string;
  description?: string | null;
  audio_url: string;
  duration_sec?: number | null;
  published_at?: string | null;
  downloaded: boolean;
  book_id?: string | null;
  created_at: string;
  updated_at: string;
}

export interface SubscribePodcastInput {
  feed_url: string;
  library_id: string;
}

export interface UpdatePodcastInput {
  auto_download?: boolean;
}

File: internal/dtos/response/podcast.go
package response

import "time"

type PodcastResponse struct {
	ID            string     `json:"id"`
	LibraryID     string     `json:"library_id"`
	FeedURL       string     `json:"feed_url"`
	Title         string     `json:"title"`
	Description   *string    `json:"description,omitempty"`
	CoverURL      *string    `json:"cover_url,omitempty"`
	Author        *string    `json:"author,omitempty"`
	AutoDownload  bool       `json:"auto_download"`
	LastCheckedAt *time.Time `json:"last_checked_at,omitempty"`
	EpisodeCount  int64      `json:"episode_count,omitempty"`
	CreatedAt     time.Time  `json:"created_at"`
	UpdatedAt     time.Time  `json:"updated_at"`
}

type PodcastEpisodeResponse struct {
	ID          string     `json:"id"`
	PodcastID   string     `json:"podcast_id"`
	GUID        string     `json:"guid"`
	Title       string     `json:"title"`
	Description *string    `json:"description,omitempty"`
	AudioURL    string     `json:"audio_url"`
	DurationSec *int64     `json:"duration_sec,omitempty"`
	PublishedAt *time.Time `json:"published_at,omitempty"`
	Downloaded  bool       `json:"downloaded"`
	BookID      *string    `json:"book_id,omitempty"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
}
